src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,133 @@
+import classNames from 'classnames';
+import { useEffect, useRef, useState } from 'react';
+import { BatchOperations } from '../../types/BatchOperations';
+import { Todo } from '../../types/Todo';
+
+interface TodoItemProps {
+  title: string;
+  id: number;
+  completed: boolean;
+  allLoading: boolean;
+  allComplete: boolean;
+  batchOperation: string | null;
+  onDeleteTodo: (todoId: number) => void;
+  onUpdateTodo: (id: number, todo: Partial<Todo>) => void;
+}
+
+export const TodoItem: React.FC<TodoItemProps> = ({
+  title,
+  id,
+  completed,
+  allLoading,
+  allComplete,
+  batchOperation,
+  onDeleteTodo,
+  onUpdateTodo,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [isEditingEnabled, setIsEditingEnabled] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleComplete = () => {
+    setIsLoading(true);
+    onUpdateTodo(id, { completed: !completed });
+  };
+
+  const handleDelete = () => {
+    setIsLoading(true);
+    onDeleteTodo(id);
+  };
+
+  const handleUpdateTitle = () => {
+    setIsEditingEnabled(false);
+
+    if (newTitle !== title) {
+      if (newTitle.trim().length > 0) {
```

Comment:
maybe you can do it this way?

```suggestion
    if (newTitle !== title && newTitle.trim().length > 0) {
```

Reply:
I structured it that way because both conditions inside require the new title to be different from the old one; to remove if nesting, I would need to write a not-so-pretty-looking thing like this:
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/55781950/f6278a51-6881-4487-928a-ba4cb1406ff1)

Reply:
The point I'm trying to make is that there's a repeated condition this way

Reply:
Though I suppose I could just check the newtitle against the title first and return if they are equal; will apply

Reply:
okay i get it, you can leave it how it was 👍